### PR TITLE
forward port 7 to 8

### PR DESCRIPTION
**Pull request by lepistone**

### CODE REVIEW

File: picking_dispatch/picking_dispatch.py [hunk 1]
```diff
@@ -182,10 +182,50 @@ def assert_start_ok(self):
                     _('Error'),
                     _('This dispatch cannot be processed until %s') % obj.date)
 
-    @api.multi
-    def action_assign_moves(self):
-        for rec in self:
-            rec.move_ids.action_assign()
+    def action_assign_moves(self, cr, uid, ids, context=None):
```
Comment (pedrobaeza) [inline picking_dispatch/picking_dispatch.py:185]: This has been changed according new API, so you should revert this

Comment (lepistone) [inline picking_dispatch/picking_dispatch.py:185]: I'd like to keep it like that. The new v7 code is a huge performance improvement for the reservation of big dispatches. The new API lines we're giving up are trivial, so I think that's a win.

Comment (pedrobaeza) [inline picking_dispatch/picking_dispatch.py:185]: OK then

**Comment by pedrobaeza** [comment]
:+1:

File: picking_dispatch/picking_dispatch.py [hunk 2]
```diff
@@ -182,10 +182,50 @@ def assert_start_ok(self):
...
+            )
+
+            if res and len(res) == 1:
+                move_obj.write(cr, uid, move_ids, {'state': 'assigned',
+                                                   'location_id': res[0][1]})
+            else:
+                # We could not reserve all the moves together, or the
+                # reservation is split among many sublocations.
+                # If that's the case, fall back to reserving one move at a time
+                move_obj.action_assign(cr, uid, move_ids)
```
Comment (yvaucher) [inline picking_dispatch/picking_dispatch.py]: Missing context propagation here

File: picking_dispatch/picking_dispatch.py [hunk 3]
```diff
@@ -182,10 +182,50 @@ def assert_start_ok(self):
...
+                [location_id],
+                product_id,
+                quantity,
+                lock=True,
+                context=context
+            )
+
+            if res and len(res) == 1:
+                move_obj.write(cr, uid, move_ids, {'state': 'assigned',
+                                                   'location_id': res[0][1]})
```
Comment (yvaucher) [inline picking_dispatch/picking_dispatch.py]: Missing context propagation here

Comment (lepistone) [inline picking_dispatch/picking_dispatch.py]: well spotted. I should fix that in v7.

Comment (lepistone) [inline picking_dispatch/picking_dispatch.py]: see #140

**Comment by lepistone** [comment]
I merged in 7.0 again now that #140 is merged.

**Comment by pedrobaeza** [comment]
:+1:

**Comment by guewen** [comment]
:+1:

**Comment by yvaucher** [comment]
:+1:

**Pull request merged**